apple/internal/transition_support.bzl

```
@@ -40,13 +40,18 @@ def _cpu_string(*, cpu, platform_type, settings):
         cpu_value = settings["//command_line_option:cpu"]
         if cpu_value.startswith("ios_"):
             return cpu_value
+        if cpu_value == "darwin_arm64":
+            return "ios_sim_arm64"
         return "ios_x86_64"
     if platform_type == "macos":
         if cpu:
             return "darwin_{}".format(cpu)
         macos_cpus = settings["//command_line_option:macos_cpus"]
         if macos_cpus:
             return "darwin_{}".format(macos_cpus[0])
+        cpu_value = settings["//command_line_option:cpu"]
+        if cpu_value.startswith("darwin_"):
```

On Intel Macs, unless `--cpu=darwin_x86_64` is passed, this defaults to `darwin`.

```suggestion
        if cpu_value.startswith("darwin"):
```

---

But would we want to return `darwin`, or fall through to the `darwin_x86_64` case? I think we want to preserve old behavior for Intel.

---

hrm yea that's kinda weird either way. I guess in that case we can continue letting it fallthrough as it was before, but we should try to eliminate that ambiguous arch identifier anyways